### DISCUSSION TITLE
Looks like you access _root->prev_sibling_c without check if it is nullptr

### DISCUSSION THREAD

**Discussion by eantcal** [zeux/pugixml#465] [General]
There is this line can be a potential defect since looks like _root->prev_sibling_c might be null. Not sure if there is anything which prevents this happening, but looks like a crash could be related to this check in one of oldest version of the library (I cannot be more specific since it is related to code I cannot share). Just in case anyone has a chance to refactor it, it would be safer and more robust checking _root->prev_sibling_c is not null either (as it is done for _root pointer in the same method).

https://github.com/zeux/pugixml/blob/9e382f98076e57581fcc61323728443374889646/src/pugixml.cpp#L5618

**Comment by zeux** [comment]
prev_sibling_c should never be null unless the node was removed from the tree (in which case it's invalid to work with the node and other functions may result in a crash).

_c refers to "cyclic": in the first child, this pointer points to the last sibling (if present) or the node itself if absent

**Reply by eantcal** [reply]
I see. Thanks, that makes sense.
